### PR TITLE
BUG: Return UTC DateTimeIndex when specified in to_datetime

### LINKED ISSUE #11934: Bug: UTC in to_datetime() not working in 0.17.1
In the case below, setting `utc=True` in pandas.to_datetime actually removes the UTC attribute from the input:

    import pandas as pd
    import pytz
    start = pd.Timestamp('2014-01-01', tz=pytz.UTC)
    end = pd.Timestamp('2014-01-03', tz=pytz.UTC)
    date_range = pd.bdate_range(start, end)
    df = pd.DataFrame({'x':1}, index=date_range)
    df.index
    # DatetimeIndex([...], dtype='datetime64[ns, UTC]', freq='B')
    pd.to_datetime(df.index, utc=True)
    # DatetimeIndex([...], dtype='datetime64[ns]', freq='B')

the last function removes the UTC attribute from the input, even though `utc` is set to `True`.

This only happens in pandas 0.17.1.  pandas 0.16.x have a different implementation and does not have this issue.

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.  Closes #11934.

### CODE REVIEW

**Comment by gfyoung** [comment]
1) Will add "what's new" entry as soon as Travis gives me a green light.
2) Where should I put a test for this bug?  There is no `test_tools.py` file AFAICT.

**Comment by jreback** [comment]
tseries/test_timeseries

look for similar tests

**Comment by gfyoung** [comment]
@jreback : All tests are passing.  Should be good to merge then.

File: pandas/tseries/tests/test_timeseries.py [hunk 1]
```diff
@@ -1143,6 +1143,18 @@ def test_to_datetime_tz_pytz(self):
...
+    def test_to_datetime_utc_is_true(self):
+        # See gh-11934
+        start = pd.Timestamp('2014-01-01', tz='utc')
+        end = pd.Timestamp('2014-01-03', tz='utc')
+        date_range = pd.bdate_range(start, end)
+        df = pd.DataFrame({'x': 1}, index=date_range)
+        date_index = pd.to_datetime(df.index, utc=True)
+
+        expected = 'datetime64[ns, UTC]'
+        out_dtype = str(date_index.dtype)
```
Comment (jreback) [inline pandas/tseries/tests/test_timeseries.py]: construct an actual DatetimeIndex as the expect d result and use assert_index_equal

you don't need to construct the DataFrame here as you are only using the index

Comment (gfyoung) [inline pandas/tseries/tests/test_timeseries.py]: Done.

**Comment by gfyoung** [comment]
@jreback : All tests are passing.  Should be good to merge then.

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**